src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java

```diff
@@ -0,0 +1,166 @@
+            + "no street name specified in the address. No suggestions names were found as there were no"
+            + "nearby nodes or ways with street name key tags.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            NO_STREET_NAME_POINT_INSTRUCTIONS, NO_STREET_NAME_EDGE_INSTRUCTIONS,
+            NO_SUGGESTED_NAMES_INSTRUCTIONS);
+    private static final String ADDRESS_STREET_NUMBER_KEY = AddressHousenumberTag.KEY;
+    private static final String POINT_STREET_NAME_KEY = AddressStreetTag.KEY;
+    private static final String EDGE_STREET_NAME_KEY = NameTag.KEY;
+    private static final String STREET_RELATION_ROLE = "street";
+    private static final String ASSOCIATED_STREET_RELATION = "associatedStreet";
+    private static final double BOUNDS_SIZE_DEFAULT = 150.0;
+
+    private final Distance boundsSize;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    public AddressPointMatchCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.boundsSize = Distance.meters(
+                (Double) configurationValue(configuration, "bounds.size", BOUNDS_SIZE_DEFAULT));
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Object is an instance of Point
+        return object instanceof Point
+                // And does not have an Associated Street Relation
+                && !hasAssociatedStreetRelation(object)
+                // And has a street number specified
+                && object.getTag(ADDRESS_STREET_NUMBER_KEY).isPresent()
+                // And if the street name key has a value of null or if the street name key is not
+                // present
+                && Strings.isEmpty(object.tag(POINT_STREET_NAME_KEY));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Point point = (Point) object;
+
+        // Get a bounding box around the Point of interest
+        final Rectangle box = point.getLocation().boxAround(boundsSize);
+
+        // Get all Points in the bounding box, remove Points that have null as their
+        // street name or do not have the street name key tag, and get a set of candidate street
+        // names
+        final Set<String> points = Iterables.stream(point.getAtlas().pointsWithin(box))
+                .filter(nearbyPoint -> nearbyPoint.getTag(POINT_STREET_NAME_KEY).isPresent())
+                .map(nearbyPoint -> nearbyPoint.tag(POINT_STREET_NAME_KEY)).collectToSet();
```

Comment:
`nearbyPoint`'s street name tag is accessed twice here if it exists. I'd suggest mapping street name tags first (via `tag` method) and then filtering out nulls later and finally collecting them into a set.